lib/mongoid/shardable.rb

```diff
@@ -53,12 +53,10 @@ def shard_key_fields
     # @return [ Hash ] The shard key selector.
     #
     # @api private
-    def shard_key_selector
-      selector = {}
-      shard_key_fields.each do |field|
-        selector[field.to_s] = send(field)
+    def shard_key_selector(prefer_persisted: false)
```

Comment:
Maybe `use_persisted` instead of `prefer_persisted`? A preference to me would indicate a fallback situation: "use A if available, otherwise use B." But in this case, it setting the option to `true` looks like it deterministically always uses A instead of B.

Reply:
I guess, in my mind, `prefer_persisted` *does* indicate the fallback situation, more than `use_persisted` does. When `true`, it will *prefer* the persisted version of the field; when that isn't possible, it will fall back to the current version.